httpclient/api_client.go

```diff
@@ -211,16 +220,35 @@ func (c *ApiClient) attempt(
 		if request.Header.Get("Content-Type") == "" && requestBody.ContentType != "" {
 			request.Header.Set("Content-Type", requestBody.ContentType)
 		}
+		// If there is a request body, wrap it to extend the request timeout while it is being read.
+		// Note: we do not wrap the request body earlier, because [http.NewRequestWithContext] performs
+		// type probing on the body variable to determine the content length.
+		if request.Body != nil && request.Body != http.NoBody {
```

Comment:
@mgyucht I needed to check for `http.NoBody` as well. The guts of `net/http` use this to signal an empty body if it is not nil... We clearly don't care about extending timeouts if there is no body, so safe to filter here.